src/occ_map/voxel_map.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 Massachusetts Institute of Technology
+#pragma once
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <stdlib.h>
+#include <math.h>
+#include <assert.h>
+
+#define NO_LCM
+
+namespace occ_map {
+
+template<class T>
+class VoxelMap {
+public:
+  //metadata
+  double xyz0[3], xyz1[3];
+  double metersPerPixel[3];
+  int dimensions[3];
+  int num_cells;
+
+#ifndef NO_LCM
+  occ_map_voxel_map_t *msg;
+#endif
+  int64_t utime;
+
+  //the actual storage arrays
+  T * data;
+
+  // Normal Constructor
+  VoxelMap<T>(const double _xyz0[3], const double _xyz1[3], const double _metersPerPixel[3], T initValue = T(),
+      bool allocate_data = true);
+
+  // Copy Constructor
+  template<class F>
+  VoxelMap<T>(const VoxelMap<F> * to_copy, bool copyData = true, T (*transformFunc)(F) = NULL);
+
+#ifndef NO_LCM
+  // Constructor from an LCM message
+  VoxelMap<T>(const occ_map_voxel_map_t * _msg);
+
+  // Constructor from a file (created with "saveToFile")
+  VoxelMap<T>(const std::string & name);
+#endif
+
+  ~VoxelMap<T>();
+
+  // set all values in the map to 0
+  void reset(T resetVal = T());
+
+  //get linear index into storage arrays
+  inline int getInd(const int ixyz[3]) const;
+  inline int getInd(const double xyz[3]) const;
```

Comment:
You should double check whether these inlines actually have an effect. I think the last time I read up on it, the compiler basically will inline whatever it wants no matter what you tell it. The `inline` keyword is only *needed* if you were to define the function inside the header file (in which case the `inline` is needed to tell the compiler only to include the definition once).